javatests/arcs/core/storage/StorageKeyParserTest.kt

```diff
@@ -53,10 +51,35 @@ class StorageKeyParserTest {
         assertThat(thrownError).isInstanceOf(IllegalArgumentException::class.java)
     }
 
+    @Test
+    fun testRegistrationRacing() = runBlocking<Unit> {
```

Comment:
Note: although this test is introduced here, I did try it on the current master, and it failed consistent, usually by the 1-200th interation.

Reply:
Sounds like a good test then!